libvcx/src/api/vcx.rs

```diff
@@ -145,6 +145,31 @@ pub extern fn vcx_open_wallet(command_handle: CommandHandle, cb: extern fn(xcomm
     error::SUCCESS.code_num
 }
 
+/// Creates agent client, which is necessary for communication with the agency. Can be called only after wallet has been opened via vcx_open_wallet.
+///
+/// #Params
+/// command_handle: command handle to map callback to user context.
+/// wallet_handle: wallet handle for the wallet to be used by the agent client (currently not used)
+/// config: config of the agent client (currently not used)
+///
+/// cb: Callback that provides error status of initialization
+///
+/// #Returns
+/// Error code as a u32
+#[no_mangle]
+pub extern fn vcx_init_agency_client(command_handle: CommandHandle, wallet_handle: WalletHandle, config: *const c_char, cb: extern fn(xcommand_handle: CommandHandle, err: u32)) -> u32 {
```

Comment:
I'd hold off from extending API just yet. Let's keep this PR internal changes and refactoring only, then create issue with new API proposal (onboarding new agent, initializing existing agent, creating wallet, opening wallet, etc. ) - then implement those APIs in separate PR.